library/java/io/envoyproxy/envoymobile/engine/EnvoyEngineImpl.java

```diff
@@ -55,13 +55,23 @@ public void terminate() {
    * @return A status indicating if the action was successful.
    */
   @Override
-  public int runWithConfig(String configurationYAML, String logLevel) {
-    try {
-      return JniLibrary.runEngine(this.engineHandle, configurationYAML, logLevel);
-    } catch (Throwable throwable) {
-      // TODO: Need to have a way to log the exception somewhere.
-      return ENVOY_FAILURE;
+  public int runWithConfig(EnvoyConfiguration envoyConfiguration, String configurationYAML,
```

Comment:
nit: please update the the above javadoc. 

Also the configurationYAML parameter can now have arguments substituted. Was not the case before.

Reply:
let me update with a blurb on the java docs